servers/zms/src/main/java/com/yahoo/athenz/zms/provider/ServiceProviderClient.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    private HttpDriver getHttpDriver(SSLContext sslContext, ServiceProviderManager serviceProviderManager) {
+
+        int maxPoolRoute = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_MAX_POOL_ROUTE, "45"));
+        int maxPoolTotal = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_MAX_POOL_TOTAL, "50"));
+        int clientRetryIntervalMs = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_RETRY_INTERVAL_MS, "1000"));
+        int clientMaxRetries = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_MAX_RETRIES, "2"));
+        int clientConnectTimeoutMs = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_CONNECT_TIMEOUT_MS, "5000"));
+        int clientReadTimeoutMs = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_READ_TIMEOUT_MS, "15000"));
+
+        return new HttpDriver.Builder("", sslContext)
+                .maxPoolPerRoute(maxPoolRoute)
+                .maxPoolTotal(maxPoolTotal)
+                .clientRetryIntervalMs(clientRetryIntervalMs)
+                .clientMaxRetries(clientMaxRetries)
+                .clientConnectTimeoutMs(clientConnectTimeoutMs)
+                .clientReadTimeoutMs(clientReadTimeoutMs)
+                .hostnameVerifier(new ProviderHostnameVerifier(serviceProviderManager))
+                .build();
+    }
+
+    private DomainDependencyProviderResponse getDependencyStatusFromProvider(ServiceProviderManager.DomainDependencyProvider domainDependencyProvider, String domain, String principal) throws IOException {
+        DomainDependencyProviderResponse domainDependencyProviderResponse = new DomainDependencyProviderResponse();
+        String url = getProviderEndpoint(domainDependencyProvider, domain);
+
+        HttpPost httpPost = new HttpPost(url);
+        httpPost.setHeader("Content-type", "application/json");
+        ProviderDependencyRequest providerDependencyRequest = new ProviderDependencyRequest(
+                "delete",
+                domain,
+                "domain",
+                domain,
+                principal,
+                domainDependencyProvider.getProvider()
+        );
+        String body = jsonMapper.writeValueAsString(providerDependencyRequest);
+        StringEntity stringEntity = new StringEntity(body);
+        httpPost.setEntity(stringEntity);
+        HttpDriverResponse httpResponse = httpDriver.doPostHttpResponse(httpPost);
+        if (httpResponse.getStatusCode() >= 500 && httpResponse.getStatusCode() <= 599) {
+            // provider error - principal should retry operation
+            domainDependencyProviderResponse.setStatus(PROVIDER_RESPONSE_DENY);
+            domainDependencyProviderResponse.setMessage("Http Status: " + httpResponse.getStatusCode() + ", error: " + httpResponse.getMessage());
+        } else if (httpResponse.getStatusCode() >= 400 && httpResponse.getStatusCode() <= 499) {
+            // TODO: Log error and send notification for athenz team on request failure
```

Comment:
Will be implemented in an upcoming pr

Reply:
EDIT: implemented. An error message was added which can be tracked by cloud watch.